### ISSUE TITLE
SPMI for Arm64 failing asserting for `argReg == argNode->GetRegNum()`

### ISSUE THREAD

**Issue by tannergooding** [dotnet/runtime#101070]
## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=643030&view=logs&jobId=80130f07-32fc-5c19-809b-35051dcc5ef5
Build error leg or test failing: SuperPMI replay windows x64 checked

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": "argReg == argNode->GetRegNum()",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

**Comment by tannergooding** [comment]
This is failing for a number of the legs (any that actually run the SPMI jobs): https://dev.azure.com/dnceng-public/public/_build?definitionId=150&_a=summary

This was likely introduced in https://github.com/dotnet/runtime/pull/100589 and is probably related to the `LoadVector128x4` or related APIs which are relatively new.

**Comment by jeffschwMSFT** [comment]
> Build analysis seems not to be matching on this... any idea why?

Build analysis has been restricted to the CI/PR pipeline for now.  There is engineering work to re-enable for our other pipelines.

**Comment by kunalspathak** [comment]
@SwapnilGaikwad

**Comment by a74nh** [comment]
The storex instructions allow for register wraparound (eg 30, 31, 0, 1). If the register allocator supports this, then the assert in [hwintrinsiccodegenarm64.cpp](https://github.com/dotnet/runtime/blob/7a5542df8b86fe207092cb72c32707a65007f974/src/coreclr/jit/hwintrinsiccodegenarm64.cpp#L836C19-L836C47) is incorrect - it needs to allow for the register to wrap back to 0.

Alternatively - if the register allocator doesn't support wrap around then my only thought is that the allocator is failing to allocate a valid sequence and silently failing.

**Comment by SwapnilGaikwad** [comment]
> then the assert in [hwintrinsiccodegenarm64.cpp](https://github.com/dotnet/runtime/blob/7a5542df8b86fe207092cb72c32707a65007f974/src/coreclr/jit/hwintrinsiccodegenarm64.cpp#L836C19-L836C47) is incorrect - it needs to allow for the register to wrap back to 0.

Indeed, we needed to wraparound the argReg - #101430 implements it.

**Comment by kunalspathak** [comment]
Fixed by https://github.com/dotnet/runtime/pull/101430
